plugins/core/src/test/kotlin/software/aws/toolkits/jetbrains/core/PluginCoreJvmBinaryCompatabilityTest.kt

```diff
@@ -0,0 +1,62 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+class PluginCoreJvmBinaryCompatabilityTest {
+    @Test
+    fun `loginSso JVM signature is stable`() {
+        // v1.0.133.0 of internal plugin
+        // $ javap -c -classpath aws-toolkit-amazonq-2024.1.jar <...>.amazonq.AmazonQConnectionService
+        //   public final void authenticateInternal$aws_toolkit_amazonq(kotlin.jvm.functions.Function1<? super java.lang.Boolean, kotlin.Unit>);
+        //       38: invokestatic  #110                // Method software/aws/toolkits/jetbrains/core/credentials/ToolkitAuthManagerKt.loginSso$default:(
+        //           Lcom/intellij/openapi/project/Project;
+        //           Ljava/lang/String;Ljava/lang/String;
+        //           Ljava/util/List;
+        //           Lkotlin/jvm/functions/Function1;
+        //           Lkotlin/jvm/functions/Function1;
+        //           Lkotlin/jvm/functions/Function0;
+        //           Lsoftware/aws/toolkits/jetbrains/core/credentials/ConnectionMetadata;
+        //           I
+        //           Ljava/lang/Object;
+        //       )Lsoftware/aws/toolkits/jetbrains/core/credentials/AwsBearerTokenConnection;
+
+        // loginSso(...)
+        val clazz = Class.forName("software.aws.toolkits.jetbrains.core.credentials.ToolkitAuthManagerKt")
+        val method = clazz.getDeclaredMethod(
+            "loginSso\$default",
+            Class.forName("com.intellij.openapi.project.Project"),
+            Class.forName("java.lang.String"),
+            Class.forName("java.lang.String"),
+            Class.forName("java.util.List"),
+            Class.forName("kotlin.jvm.functions.Function1"),
+            Class.forName("kotlin.jvm.functions.Function1"),
+            Class.forName("kotlin.jvm.functions.Function0"),
+            Class.forName("software.aws.toolkits.jetbrains.core.credentials.ConnectionMetadata"),
+            // can't request primitive type using reflection
+            Integer.TYPE,
+            Class.forName("java.lang.Object"),
```

Comment:
Where are these parameter types coming from? [The method signature](https://github.com/aws/aws-toolkit-jetbrains/blob/1d081527aee07dc18b9ad22ab97db6653b64be9d/plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/ToolkitAuthManager.kt#L112-L121) ends with `ConnectionMetadata`.

Reply:
`*$default` is a synthetic method generated by the Kotlin compiler. Based on the bytecode extracted from the internal plugin jar, this is what is being invoked.